cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js

```diff
@@ -217,49 +234,98 @@ describe('Unit testing ListController', function () {
 
         beforeEach(function () {
             deleteCrosslistingDeferred = $q.defer();
-            scope.dtInstance = jasmine.createSpyObj('dtInstance', ['reloadData']);
             spyOn(scope, 'deleteCrosslisting').and
                 .returnValue(deleteCrosslistingDeferred.promise);
             spyOn(scope, 'handleAjaxErrorResponse');
+            spyOn(scope, 'setOperationInProgress');
         });
 
         it('calls deleteCrosslisting with the correct id', function() {
             scope.removeCrosslisting(1, 2, 3);
             expect(scope.deleteCrosslisting).toHaveBeenCalledWith(1);
         });
 
-        it('shows failure message and cleans up on failure', function() {
+        it('shows failure message and reloads even on failure', function() {
             scope.removeCrosslisting(1, 1, 2);
 
-            expect(scope.operationInProgress).toBe('remove');
+            expect(scope.setOperationInProgress).toHaveBeenCalledWith('remove');
             expect(scope.message).toBe(null);
 
             deleteCrosslistingDeferred.reject('failure!');
             scope.$digest();
 
-            expect(scope.operationInProgress).toBeNull();
+            // cleanup performed dtInstance.reloadData()
             expect(scope.dtInstance.reloadData).toHaveBeenCalled();
+
             expect(scope.handleAjaxErrorResponse).toHaveBeenCalled();
             expect(scope.message.alertType).toBe('danger');
         });
 
-        it('shows success message and cleans up on success', function() {
+        it('shows success message and reloads on success', function() {
             scope.removeCrosslisting(1, 1, 2);
 
-            expect(scope.operationInProgress).toBe('remove');
+            expect(scope.setOperationInProgress).toHaveBeenCalledWith('remove');
             expect(scope.message).toBe(null);
 
             deleteCrosslistingDeferred.resolve('success!');
             scope.$digest();
 
-            expect(scope.operationInProgress).toBeNull();
```

Comment:
The actual `scope.operationInProgress` modifications are now wrapped inside `scope.setOperationInProgress()` and `scope.dtInstance.reloadData`, both of which we've mocked, so the value won't changed in this test case. Instead we'll have to assume that those two methods did their jobs... it's pretty convoluted, isn't it.